src/main/java/InfrastructureManager/MatchMaking/MatchMaker.java

```diff
@@ -0,0 +1,84 @@
+package InfrastructureManager.MatchMaking;
+
+import InfrastructureManager.*;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+public class MatchMaker implements MasterInput, MasterOutput {
+
+    private MatchMakingAlgorithm algorithm;
+    private final ObjectMapper mapper;
+    private String command = "";
+
+    public MatchMaker() {
+        this.algorithm = new RandomMatchMaking(); // For now
+        this.mapper = new ObjectMapper();
+    }
+
+    public void setAlgorithm(MatchMakingAlgorithm algorithm) {
+        this.algorithm = algorithm;
+    }
+
+    private void assign (EdgeClient client) {
+        EdgeNode node = this.algorithm.match(client);
+        if (node == null) {
+            System.out.println("No node to assign");
+        } else {
+            Master.getInstance().mapClientNode(client, node);
+            command = "give_node " + client.getId();
+        }
+        System.out.println("Client received : " + client.getId() + " Server assigned : " + node.getId());
+    }
+
+    @Override
+    public void out(String response) throws IllegalArgumentException {
+        String[] command = response.split(" ");
+        if (command[0].equals("matchMaker")) {
+            try {
+                switch (command[1]){
+                    case "register_client" :
+                        registerClient(command[2]);
+                        break;
+                    case "register_node" :
+                        registerNode(command[2]);
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Invalid command for MatchMaker");
+                }
+            } catch (IndexOutOfBoundsException e) {
+                throw new IllegalArgumentException("Arguments missing for command - MatchMaker");
+            }
+        }
+    }
+
+    private void registerClient(String clientAsString) {
+        try {
+            EdgeClient client = this.mapper.readValue(clientAsString, EdgeClient.class);
+            Master.getInstance().addClient(client);
+            assign(client); //Perform MatchMaking
```

Comment:
assigning a node to a client should be a separate command, remove it from here and add a new command to out().